RASP_support/REPL.py

```diff
@@ -219,31 +216,34 @@ def loadFile(self,ast,calling_env=None):
 			with open(filename,"r") as f:
 				prev_example_settings = self.show_sequence_examples, self.show_selector_examples
 				self.show_sequence_examples, self.show_selector_examples = False, False
-				self.run(fromfile=f,env = Environment(name=libname,parent_env=self.base_env,stealing_env=calling_env),store_prints=True)
-				self.filter_and_dump_prints() 
+				self.run(fromfile=f,env=Environment(name=libname,parent_env=self.base_env,stealing_env=calling_env))
 				self.show_sequence_examples, self.show_selector_examples = prev_example_settings
 		except FileNotFoundError:
 			raise LoadError("could not find file: "+filename)
 
-	def get_tree(self,fromfile=None):
-		try:
-			return LineReader(fromfile=fromfile).get_input_tree()
-		except AntlrException as e:
-			print("\t!! antlr exception:",e.msg,"\t-- ignoring input")
-		return None
-
-	def run_given_line(self,line):
+	def run_tree(self,fromfile,fromline,env):
 		try:
-			tree = LineReader(given_line=line).get_input_tree()
+			tree = LineReader(fromfile=fromfile,fromline=fromline).get_input_tree()
 			if isinstance(tree,Stop):
-				return None
-			rp = self.evaluate_tree(tree)
-			if isinstance(rp,LazyPrint):
-				rp.print() # error messages get raised, but ultimately have to be printed somewhere if not caught? idk
```

Comment:
do you mean this line?

Reply:
no, what is now line 232 (and was lines 244-245). Basically, the new function run_tree has one way of dealing with antlr exceptions (the 'normal' way for while the REPL is running), but it is replacing the function run_given_line which was originally just for REPL initiation, and which had an extra print if it got an antlr exception (noting which initiating line failed). I don't really mind, I was just wondering if it was a deliberate choice. 

independently, how did you put the diff inside the comment?

Reply:
Oh, I guess I didn't get the difference between initialization ANTLR errors and code ANTRL error messages, the code is new to me. I am also happy with anything that you do with this :)

For code comments, go to the top of this github pull request page, switch tabs from "conversation" to "commit" at the top, hover over line, press [+], add comment.
i think, you can also do full "code reviews" in the sense that you go line by line through the changes and comment everything you find comment-worthy.

Reply:
and eventually mark it as resolved ↓